src/containers/StreamingSettingsPanel.tsx

```diff
@@ -0,0 +1,301 @@
+import { useState } from "react";
+import { DeviceIdToStream, StreamingDeviceSelector, mockStreamNames } from "../components/StreamingDeviceSelector";
+import { useLocalStorageState, useLocalStorageStateString, useLocalStorageStateArray } from "../components/LogSelector";
+import { TrackEncoding } from "@jellyfish-dev/membrane-webrtc-js";
+import { showToastError } from "../components/Toasts";
+import { createStream } from "../utils/createMockStream";
+import { getUserMedia } from "@jellyfish-dev/browser-media-utils";
+import { createMockAudio } from "../utils/createMockAudio";
+import { DEFAULT_TRACK_METADATA } from "./Client";
+export type DeviceInfo = {
+  id: string;
+  type: string;
+};
+
+type PanelProps = {
+  name: string;
+  status: string;
+  setSimulcast: (isActive: boolean) => void;
+  simulcast: boolean;
+  trackMetadata: string | null;
+  setTrackMetadata: (value: string | null) => void;
+  maxBandwidth: string | null;
+  setMaxBandwidth: (value: string | null) => void;
+  attachMetadata: boolean;
+  setAttachMetadata: (value: boolean) => void;
+  selectedDeviceId: DeviceInfo | null;
+  setSelectedDeviceId: (data: DeviceInfo | null) => void;
+  activeStreams: DeviceIdToStream | null;
+  setActiveStreams: (setter: ((prev: DeviceIdToStream | null) => DeviceIdToStream) | DeviceIdToStream | null) => void;
+  currentEncodings: TrackEncoding[];
+  setCurrentEncodings: (value: TrackEncoding[]) => void;
+  addAudioTrack: (stream: MediaStream) => void;
+  addVideoTrack: (stream: MediaStream) => void;
+};
+
+const emojiIdToIcon = (emojiId: string) => {
+  switch (emojiId) {
+    case "HEART_STREAM":
+      return "💜";
+    case "FROG_STREAM":
+      return "🐸";
+    case "ELIXIR_STREAM":
+      return "🧪";
+    case "OCTOPUS_STREAM":
+      return "🐙";
+    default:
+      return "N/A";
+  }
+};
+
+const checkJSON = (s: string) => {
+  s = s.trim();
```

Comment:
please don't use single letter variable names and please don't reassign the parameter